internal/consensus/pbts_test.go

```diff
@@ -1,56 +1,299 @@
+	propBlock, partSet := p.observedState.createProposalBlock()
+	bid := types.BlockID{Hash: propBlock.Hash(), PartSetHeader: partSet.Header()}
+	ensureProposal(t, p.ensureProposalCh, p.currentHeight, p.currentRound, bid)
+	ensurePrevote(t, p.ensureVoteCh, p.currentHeight, p.currentRound)
+	signAddVotes(p.observedState, tmproto.PrevoteType, p.chainID, bid, p.otherValidators...)
+
+	signAddVotes(p.observedState, tmproto.PrecommitType, p.chainID, bid, p.otherValidators...)
+	ensurePrecommit(t, p.ensureVoteCh, p.currentHeight, p.currentRound)
+
+	ensureNewBlock(t, p.blockCh, p.currentHeight)
+	p.currentHeight++
+	incrementHeight(p.otherValidators...)
+}
+
+func (p *pbtsTestHarness) height2(t *testing.T) heightResult {
+	signer := p.otherValidators[0].PrivValidator
+	return p.nextHeight(t, signer, p.s.height2ProposalDeliverTime, p.s.height2ProposedBlockTime, time.Now())
+}
+
+func (p *pbtsTestHarness) nextHeight(t *testing.T, proposer types.PrivValidator, deliverTime, proposedTime, nextProposedTime time.Time) heightResult {
+	p.validatorClock.On("Now").Return(nextProposedTime).Times(8)
+	pubKey, err := p.observedValidator.PrivValidator.GetPubKey(context.Background())
+	assert.NoError(t, err)
+	resultCh := collectResults(t, p.observedState.eventBus, pubKey.Address())
+
+	ensureNewRound(t, p.roundCh, p.currentHeight, p.currentRound)
+
+	b, _ := p.observedState.createProposalBlock()
+	b.Height = p.currentHeight
+	b.Header.Height = p.currentHeight
+	b.Header.Time = proposedTime
+
+	k, err := proposer.GetPubKey(context.Background())
+	assert.NoError(t, err)
+	b.Header.ProposerAddress = k.Address()
+	ps := b.MakePartSet(types.BlockPartSizeBytes)
+	bid := types.BlockID{Hash: b.Hash(), PartSetHeader: ps.Header()}
+	prop := types.NewProposal(p.currentHeight, 0, -1, bid)
+	tp := prop.ToProto()
+
+	if err := proposer.SignProposal(context.Background(), p.observedState.state.ChainID, tp); err != nil {
+		t.Fatalf("error signing proposal: %s", err)
+	}
+
+	time.Sleep(time.Until(deliverTime))
+	prop.Signature = tp.Signature
+	if err := p.observedState.SetProposalAndBlock(prop, b, ps, "peerID"); err != nil {
+		t.Fatal(err)
+	}
+	ensureProposal(t, p.ensureProposalCh, p.currentHeight, 0, bid)
+
+	ensurePrevote(t, p.ensureVoteCh, p.currentHeight, p.currentRound)
+	signAddVotes(p.observedState, tmproto.PrevoteType, p.chainID, bid, p.otherValidators...)
+
+	signAddVotes(p.observedState, tmproto.PrecommitType, p.chainID, bid, p.otherValidators...)
+	ensurePrecommit(t, p.ensureVoteCh, p.currentHeight, p.currentRound)
+
+	p.currentHeight++
+	incrementHeight(p.otherValidators...)
+	res := <-resultCh
+	return res
+}
+
+func collectResults(t *testing.T, eb *types.EventBus, address []byte) <-chan heightResult {
+	t.Helper()
+	resultCh := make(chan heightResult)
+	voteSub, err := eb.SubscribeUnbuffered(context.Background(), "voteSubscriber", types.EventQueryVote)
+	assert.NoError(t, err)
+	go func() {
+		res := heightResult{}
```

Comment:
Per https://github.com/uber-go/guide/blob/master/style.md#use-var-for-zero-value-structs
```suggestion
		var res heightResult
```

Reply:
TIL